### PR TITLE
Update to Lisk Core 1.0.0 API - Closes #341

### LINKED ISSUE #341: Update Lisk Nano to work with Lisk 1.0.0 API
Lisk 1.0.0 will have some breaking changes, so we need to update.

### PR DESCRIPTION
**Pull request by slaweet**
### What was the problem?

Lisk Nano used Lisk Core 0.9.X API

### How did I fix it?

Updated to Lisk Core 1.0.0 API

Changes include the following:
- some endpoint names changed (e.g. `/api/accounts/delegates` to `/api/votes`)
- each response now contains the main content in `data` object, before the name of the object depended on the endpoint (e.g. `transactions`, `delegates`)
- `account.secondSignature` property was removed. `account.secondPublicKey` can be used instead
- `delegate. rate` property was removed. `delegate.rank` can be used instead
- `delegate.address` and `delegate.publicKey` was moved to `delegate.account.address` and `delegate.account.publicKey`

There were some issues in Lisk Core:

Every thing works completely with latest changes  of `lisk core` witch contains in `development` branch, but there is no release for some of these features. So I had to suspend some of e2e tests.

--  infinite scroll in all lists are broken. there is a separated task to fix them #1065 
### How to test it?

1. Checkout `development` branch of lisk core
2. Then run lisk nano and check all features

### Review checklist

- The PR solves #341
- All new code is covered with unit tests
- All new features are covered with e2e tests
- All new code follows best practices

### CODE REVIEW

**Comment by webmaster128** [comment]
Could you rebase this PR branch onto 1.3.4 or development? Right now this is based on 1.3.2 so all the fixes between 1.3.2 and 1.3.4 are not available to Betanet testers

**Comment by slaweet** [comment]
@webmaster128 I just updated this branch to contain all the changes from development. There are still some fixes to be made by @yasharAyari next week when he's back from his vacation.

**Comment by webmaster128** [comment]
Thanks!

**Comment by slaweet** [comment]
I found the following issues:
- Websocket connection to https://betanet.lisk.io throws 400
- Delegates are not sorted by rank
- Vote API error is not handled, results in: `api_resource.js:65 Uncaught (in promise) Error: Status 409 : Failed to add vote, delegate "lisksnake" already voted for`
- Verify message should show a message "Message verified"/ "Message not verified" instead of "true"/ nothing
- Extra slash at the end of custom node URL (e.g. https://betanet.lisk.io/) should allow me to log it.
- The app fails to start with saved accounts. It throws: `Error: APIClient requires nodes for initialization.`
@yasharAyari you can fix them in this PR od create a follow-up.

File: src/actions/peers.js [hunk 1]
```diff
@@ -31,20 +32,22 @@ export const activePeerSet = data =>
...
     if (config.testnet === undefined && config.port !== undefined) {
       config.testnet = config.port === '7000';
     }
     if (config.custom) {
-      getNethash(Lisk.api(config)).then((response) => {
-        config.testnet = response.nethash === netHashes.testnet;
-        if (!config.testnet && response.nethash !== netHashes.mainnet) {
-          config.nethash = response.nethash;
+      const getNethash = new Lisk.APIClient(config.nodes, config.nethash, {});
+      getNethash.node.getConstants().then((response) => {
```
Comment (slaweet) [inline src/actions/peers.js]: `getNethash` is not describing what is stored in the variable, it is more something like `apiClientInstance`

**Review by slaweet — COMMENTED** [review]
One more thing:
- all API calls used to dispatch event `loadingStarted` before and `loadingFinished` after to show/hide the loading bar.

File: src/store/middlewares/login.js [hunk 2]
```diff
@@ -24,15 +25,17 @@ const loginMiddleware = store => next => (action) => {
     address,
   };
   const { activePeer } = action.data;
-
+  loadingStarted('loginMiddleware');
+  loadingFinished('loginMiddleware');
```
Comment (slaweet) [inline src/store/middlewares/login.js]: `loadingFinished` looks like extra here

**Review by slaweet — COMMENTED** [review]
In general `loadingFinished` should be not only in all `then` callbacks but also in `catch` callbacks. Otherwise, every time something fails, the loading bar stays forever.

**Comment by webmaster128** [comment]
Is this branch supposed to work with beta6? I get an error 400 when loading http://node01.betanet.lisk.prolina.org:5000/api/transactions?senderIdOrRecipientId=6237734014301098166L&limit=25&offset=0&sort=timestamp%3Adesc

**Comment by slaweet** [comment]
@webmaster128 That parameter will be fixed in beta7 https://github.com/LiskHQ/lisk/pull/1920

**Review by slaweet — COMMENTED** [review]
Now it's great. Good job, Yashar.

I cannot approve the PR, because I opened it in the first place. Feel free to approve it yourself.

**Review by yasharAyari — APPROVED** [review]

**Pull request merged**